homeassistant/components/insteon/__init__.py

```diff
@@ -151,12 +144,21 @@ def async_setup(hass, config):
     ip_port = conf.get(CONF_IP_PORT)
     username = conf.get(CONF_HUB_USERNAME)
     password = conf.get(CONF_HUB_PASSWORD)
+    hub_version = conf.get(CONF_HUB_VERSION)
     overrides = conf.get(CONF_OVERRIDE, [])
     x10_devices = conf.get(CONF_X10, [])
     x10_all_units_off_housecode = conf.get(CONF_X10_ALL_UNITS_OFF)
     x10_all_lights_on_housecode = conf.get(CONF_X10_ALL_LIGHTS_ON)
     x10_all_lights_off_housecode = conf.get(CONF_X10_ALL_LIGHTS_OFF)
 
+    # Set the default value of the IP port based on Hub version if not
+    # explicitly set
+    if not ip_port:
```

Comment:
Extract this into a config validator function, and use it in the config schema.

Reply:
OK, this was a good exercise. Figured it out but it was not obvious. Thanks for pushing me to do the right thing. You are really good at helping average developers like myself get to the right solution. I really appreciate it.